iac-extensions/jvm-framework-config/src/main/java/org/sonar/iac/jvmframeworkconfig/plugin/visitors/JvmFrameworkConfigHighlightingVisitor.java

```diff
@@ -39,6 +44,15 @@ protected void languageSpecificHighlighting() {
         highlight(tree.key(), KEYWORD);
       }
     });
+    register(Profile.class, (InputFileContext ctx, Profile tree) -> {
```

Comment:
Did you consider adding the `originalYamlTree` as a child and simply adding the `YamlHighlightingVisitor` to the Visitors?
 That way we can remove the `retrieveAllChildrenMethod`

Reply:
Good suggestion, I didn't think about that. However, there is one problem: in this approach, comments will be present in the tree twice, and highlighting will break.

Reply:
Mh, I didn't think about that. Either way there are present twice. Maybe we should clean them from one tree regardless?

Reply:
Cleaning them out of the YAML tree seems hacky; especially since every YamlTree node implements HasComments. Maybe, a solution with highlighting comments only from a properties file in this visitor will work. Essentially, it will turn it from JvmConfigHighlightingVisitor to PropertiesHighlightingVisitor, but that doesn't seem like a problem

Reply:
No, looks like it won't work without too many changes. Each syntax highlighting visitor tries to save highlighting in `after()` method, even if highlighting is empty. And saving highlighting twice for the same file is not supported by the plugin API

Reply:
As discussed, let's take the original approach of your PR. I believe it adds the least complexity of any solution.